### PR TITLE
Added info for <codenew> shortcode in style guide

### PR DESCRIPTION
**Pull request by dipesh-rawat**
Added documentation for `{{ <codenew> }}` shortcode in style guide.

Fixes #18231

### CODE REVIEW

File: content/en/docs/contribute/style/style-guide.md [hunk 1]
```diff
@@ -408,6 +408,33 @@ The output is:
 
 {{< kat-button >}}
 
+### Codenew
```
Comment (sftim) [inline content/en/docs/contribute/style/style-guide.md:411]:
```suggestion
### Source code files
```

**Review by sftim — COMMENTED** [review]
I would document this in https://kubernetes.io/docs/contribute/style/hugo-shortcodes/ and then have a _link_ to the next text in the style guide.

**Comment by dipesh-rawat** [comment]
> I would document this in https://kubernetes.io/docs/contribute/style/hugo-shortcodes/ and then have a _link_ to the next text in the style guide.

@sftim Thanks for taking time for review and giving feedback. As suggested, I have transferred the content over to [Custom Hugo Shortcodes](https://kubernetes.io/docs/contribute/style/hugo-shortcodes) page and added the link to this page in style guide page under the [What's next](https://kubernetes.io/docs/contribute/style/style-guide/#what-s-next) section.

File: content/en/docs/contribute/style/hugo-shortcodes/index.md [hunk 2]
```diff
@@ -271,6 +271,33 @@ Renders to:
 {{< tab name="JSON File" include="podtemplate.json" />}}
 {{< /tabs >}}
 
+### Source code files
+
+You can use `{{</* codenew */>}}` shortcode to embed the contents of file in a code block to allow users to download or copy its content to their clipboard. This shortcode is used when the contents of the sample file is generic and reusable, and you want the users to try it out themselves.
+
+This shortcode takes in two named parameters: `language` and `file`. The mandatory parameter `file` is used to specify the path to the file being displayed. The optional parameter `language` is used to specify the programming language of the file. If `language` parameter is not provided, the shortcode will attempt to guess the language based on the file extension.
```
Comment (bradtopol) [inline content/en/docs/contribute/style/hugo-shortcodes/index.md:278]:
```suggestion
This shortcode takes in two named parameters: `language` and `file`. The mandatory parameter `file` is used to specify the path to the file being displayed. The optional parameter `language` is used to specify the programming language of the file. If the `language` parameter is not provided, the shortcode will attempt to guess the language based on the file extension.
```

**Comment by bradtopol** [comment]
Thank you @dipesh-rawat for your contribution.  I found a few minor typos to address. Once these are addressed this will be ready to merge.

**Comment by dipesh-rawat** [comment]
> Thank you @dipesh-rawat for your contribution. I found a few minor typos to address. Once these are addressed this will be ready to merge.

Thanks for taking time for review and feedback. I have taken in suggested changes.

**Deploy preview:**
[https://deploy-preview-39764--kubernetes-io-main-staging.netlify.app/docs/contribute/style/hugo-shortcodes/#source-code-files](https://deploy-preview-39764--kubernetes-io-main-staging.netlify.app/docs/contribute/style/hugo-shortcodes/#source-code-files)

https://deploy-preview-39764--kubernetes-io-main-staging.netlify.app/docs/contribute/style/style-guide/#what-s-next

**Review by sftim — COMMENTED** [review]
/label tide/merge-method-squash
/lgtm
/approve

**Pull request merged**